_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -358,19 +358,21 @@ mfxDefaultAllocatorVAAPI::AllocFramesHW(
         else
         {
             VAContextID context_id = request->AllocId;
-            mfxU32 codedbuf_size;
+            mfxU32 codedbuf_size, codedbuf_num;
             VABufferType codedbuf_type;
 
             if (fourcc == MFX_FOURCC_VP8_SEGMAP)
             {
-                codedbuf_size = request->Info.Width * request->Info.Height;
+                codedbuf_size = request->Info.Width;
+                codedbuf_num = request->Info.Height;
                 codedbuf_type = (VABufferType)VAEncMacroblockMapBufferType;
```

Comment:
no need in explicit cast, ```VAEncMacroblockMapBufferType``` is part of ```VABufferType``` libva enum already
https://github.com/intel/libva/blob/f2ddc03d0b8f6ba3bb143a086687f1ad386046c6/va/va.h#L1592